nvi-rest/src/test/java/no/sikt/nva/nvi/rest/upsert/UpdateNviCandidateStatusHandlerTest.java

```diff
@@ -167,31 +225,33 @@ void shouldUpdateApprovalStatus(ApprovalStatus oldStatus, ApprovalStatus newStat
     }
 
     @ParameterizedTest(name="shouldResetFinalizedValuesWhenUpdatingStatusToPending from old status {0}")
-    @EnumSource(value = ApprovalStatus.class, names = {"REJECTED", "APPROVED"})
+    @EnumSource(value = ApprovalStatus.class, names = {STATUS_REJECTED, STATUS_APPROVED})
     void shouldResetFinalizedValuesWhenUpdatingStatusToPending(ApprovalStatus oldStatus) throws IOException {
-        var institutionId = randomUri();
-        var candidate = upsert(createUpsertCandidateRequest(institutionId).build());
-        candidate.updateApproval(createStatusRequest(oldStatus));
+        var candidate = upsert(createUpsertCandidateRequest(DEFAULT_TOP_LEVEL_INSTITUTION_ID).build());
+        candidate.updateApprovalStatus(createStatusRequest(oldStatus), mockOrganizationRetriever);
         var newStatus = ApprovalStatus.PENDING;
-        var request = createRequest(candidate.getIdentifier(), institutionId, newStatus);
+        var request = createRequest(candidate.getIdentifier(), DEFAULT_TOP_LEVEL_INSTITUTION_ID, newStatus);
         handler.handleRequest(request, output, context);
         var response = GatewayResponse.fromOutputStream(output, CandidateDto.class);
         var candidateResponse = response.getBodyObject(CandidateDto.class);
         var actualApproval = candidateResponse.approvals().getFirst();
         assertThat(actualApproval.finalizedBy(), is(nullValue()));
         assertThat(actualApproval.finalizedDate(), is(nullValue()));
-        assertThat(actualApproval.status(), is(equalTo(ApprovalStatusDto.NEW)));
+        assertThat(actualApproval.status(), is(equalTo(ApprovalStatusDto.PENDING)));
```

Comment:
Note that this changes the assert from `status=NEW` to `status=PENDING`. I believe it was an error in the test setup that made it possible to get `status=NEW`, which is now fixed (by always including the institution ID in the update request).